GTG/gtk/editor/recurring_menu.py

```diff
@@ -92,25 +90,26 @@ def update_task(self, enable=True):
 
     def update_header(self):
         """ Updates the header anytime a term is selected """
-        formated_term = self.selected_recurring_term
         if self.is_term_set():
-            if formated_term.isdigit():
-                if len(formated_term) <= 2 :
-                    day = datetime.strptime(f'{formated_term}', '%d').strftime('%d')
-                    formated_term = _('{day} of the Month')
-                    formated_term = formated_term.format(day=day)
-                else:
-                    formated_term = datetime.strptime(f'{formated_term[:2:]}-{formated_term[2::]}', '%m-%d').strftime('%B %d')
-            elif formated_term == 'week':
-                formated_term = datetime.today().strftime('%A')
-            elif formated_term == 'month':
-                formated_term = _('{day} of the Month')
-                formated_term = formated_term.format(day=datetime.today().strftime('%d'))
-            elif formated_term == 'year':
-                formated_term = datetime.today().strftime('%B %d')
+            if self.selected_recurring_term.isdigit():
+                if len(self.selected_recurring_term) <= 2 : # Recurring monthly from selected date
+                    self.title.set_markup(_('Every <b>{month_day} of the month</b>').format(month_day=datetime.strptime(f'{self.selected_recurring_term}', '%d').strftime('%d')))
+                else: # Recurring yearly from selected date
+                    self.title.set_markup(_('Every <b>{year_date}</b>').format(year_date=datetime.strptime(f'{self.selected_recurring_term[:2:]}-{self.selected_recurring_term[2::]}', '%m-%d').strftime('%B %d')))
+            elif self.selected_recurring_term == 'day': # Recurring daily
+                self.title.set_markup(_('Every <b>day</b>'))
+            elif self.selected_recurring_term == 'other-day': # Recurring every other day
+                self.title.set_markup(_('Every <b>other day</b>'))
+            elif self.selected_recurring_term == 'week': # Recurring weekly from today
+                self.title.set_markup(_('Every <b>{week_day}</b>').format(week_day=datetime.today().strftime('%A')))
+            elif self.selected_recurring_term == 'month': # Recurring monthly from today
+                self.title.set_markup(_('Every <b>{month_day} of the month</b>').format(month_day=datetime.today().strftime('%d')))
+            elif self.selected_recurring_term == 'year': # Recurring yearly from today
+                self.title.set_markup(_('Every <b>{year_date}</b>').format(year_date=datetime.today().strftime('%B %d')))
+            else: # Recurring weekly from selected week day
+                self.title.set_markup(_('Every <b>{week_day}</b>').format(week_day=self.selected_recurring_term))
```

Comment:
From my testing, if you set a week (e. g. "Friday") and then switch to german (with my currently not released updated translation), it'll then say "Jeden Friday" instead of "Jeden Freitag", because it seems it stores an once localized value rather an language neutral value and translating it later.
This doesn't seem to be good practice in my opinion.
An workaround could be `_(self.selected_recurring_term)`, but doesn't fix when switching from non-english language to an another language.

Reply:
Shouldn't day and month names be also translated? Like `_(datetime.today().strftime('%A'))`. 

---

For self.selected_recurring_term, here's what we can do:
1. Set a widget name for each week day and month. The widget name should be the english word. So for instance, the widget that has an id "mon_button" would have a name "Monday". The input for the widget name can be found in glade, in the _Common_ tab.

2. In the callback function (`set_recurring_term_week_day(self, widget)`) you get the widget. So you can do: `widget.name' to the recurring term to store. 

What do you think?
This way we make sure recurring term is always saved in english, and we can do `_(self.selected_recurring_term)`

Reply:
> Shouldn't day and month names be also translated? Like `_(datetime.today().strftime('%A'))`.

No, [from the strftime documentation](https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior):
> | Directive | Meaning | Example |
> | -- | -- | -- |
> | `%A` | Weekday as locale’s full name. | Sunday, Monday, …, Saturday (en_US); Sonntag, Montag, …, Samstag (de_DE) |

> ---
> For self.selected_recurring_term, here's what we can do:
> 
> 1. Set a widget name for each week day and month. The widget name should be the english word. So for instance, the widget that has an id "mon_button" would have a name "Monday". The input for the widget name can be found in glade, in the _Common_ tab.
> 2. In the callback function (`set_recurring_term_week_day(self, widget)`) you get the widget. So you can do: `widget.name' to the recurring term to store.
> 
> 
> What do you think?
> This way we make sure recurring term is always saved in english, and we can do `_(self.selected_recurring_term)`

I assume this is just for the week days. I also didn't really examined how it determines on how to set it, but it sounds an possible solution, it seems you don't "translate" the widget name.
The only worry I would have is that the (de-)serialization code would make sure that for example it doesn't get set to the lowercase version (like "monday" instead of "Monday") or something.